include/cynq/datamover.hpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2023
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ *         Diego Arturo Avila Torres <diego.avila@uned.cr>
+ *
+ */
+#pragma once
+#include <memory>
+
+#include "cynq/enums.hpp"
+#include "cynq/status.hpp"
+
+namespace cynq {
+class IDataMover {
+ public:
+  virtual ~IDataMover() {}
+  /**
+   * @brief GetBuffer method
+   * Retrieves the buffer used for mapping the data to memory.
+   *
+   * @param size Size in bytes of the buffer.
+   *
+   * @param type One of the values in the MemoryType enum class
+   * present in the enums.hpp file
```

Comment:
Elaborate more... Remember when you complain when reading inaccurate documentation.